src/lib/media-cache.ts

```
@@ -306,7 +306,34 @@ export async function getCachedFolders(prefix: string = ''): Promise<Folder[]> {
       throw new Error(data.error);
     }
 
-    const folders = data.folders || [];
+    const folderNames = data.folders || [];
+    const folderDetails = data.folderDetails || [];
+    
+    // Process the API response into proper Folder objects
+    const folders: Folder[] = folderNames.map((folderName: string, index: number) => {
+      // Use the actual folder name from the API, not a fallback
+      const actualFolderName = folderName && folderName.trim() ? folderName.trim() : `unnamed-folder-${index}`;
+      
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)